packages/devtools_app/test/layout_explorer/layout_explorer_serialization_delegate.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+/// This class is merely a duplicate of the InspectorSerializationDelegate inside the `assets/scripts/inspector_polyfill_script.dart`.
+/// This delegate is used for testing Widget Diagnostics Node so that we don't have to create manual JSON each time we want to create new test cases.
+/// TODO(adalberht): Ask Jacob on what's the better solution on code reuse between the Layout Explorer polyfillscripts and the code inside test package?
```

Comment:
I wanted to reuse some code on the polyfill scripts but I'm not sure on how to do that properly without cloning/copy-pasting them.
This is useful so that our unit tests can just generate the Layout Explorer Remote Diagnostics Node based on actual widgets instead of us creating JSON data for each tests cases that we want to create.
But let me know if that's not preferrable! To be honest writing it this way is way makes writing the tests more concise and pleasant :D

Reply:
Clever idea! Lets land like this and we can add a build script to copy the code so it can be kept completely in sync. The other way we could solve this is by tests communicating with a remote Dart vm like in the inspector_service tests. That has the downside that the tests are slower to run.

Reply:
Awesome, thanks, Jacob! Alright then, I agree maybe let's land like this and l will try to look on the alternative testing methods with remote Dart VM that you mentioned